lib/embulk/input/mixpanel_api/client.rb

```diff
@@ -76,6 +78,58 @@ def export_for_small_dataset(params = {})
           raise ConfigError.new "#{params["from_date"]}..#{latest_tried_to_date} has no record."
         end
 
+        def send_brief_checked_jql_script(params = {})
+          retryer.with_retry do
+            Embulk.logger.info "Sending brief check request to #{@endpoint}"
+            response = httpclient.post(@endpoint, query_string(params))
+            body = response.body
+
+            case response.code
+            when 400..499
+              if response.code == 400
+                begin
+                  return JSON.parse(body)
+                rescue =>e
+                  raise Embulk::DataError.new(e.message)
+                end
+              end
+              if response.code == 429
+                # [429] {"error": "too many export requests in progress for this project"}
+                raise RuntimeError.new("[#{response.code}] #{body} (will retry)")
+              end
+              raise ConfigError.new("[#{response.code}] #{body}")
+            when 500..599
+              raise RuntimeError.new("[#{response.code}] #{body}")
+            end
+          end
+        end
+
+        def send_jql_script(params = {})
+          retryer.with_retry do
+            response =  request_jql(params)
+
+            handle_error_for_JQL(response, response.body)
+
+            begin
+              return JSON.parse(response.body)
+            rescue =>e
+              raise Embulk::DataError.new(e)
+            end
+          end
+        end
+
+        def send_jql_script_small_dataset(params = {})
+          retryer.with_retry do
+            response = request_jql(params)
+            handle_error(response, response.body)
+            begin
+              return JSON.parse(response.body)[0..SMALL_NUM_OF_RECORDS - 1]
+            rescue =>e
+              raise Embulk::DataError.new(e.message)
+            end
+          end
```

Comment:
I don't really understand, 🤔 does this practically save us any running time/traffic?

Reply:
The reason I want to separate the guess,  preview with run. They should  fail fast and don't wait for the retry timeout. 
The code updated. Please help to doublecheck on that.